File: docs/assets/gifs/screenshots/capture_search.py

import asyncio
from pathlib import Path

from playwright.async_api import async_playwright


async def capture_screenshots():
    screenshots_dir = Path(__file__).resolve().parent
    
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=False)
        context = await browser.new_context(viewport={"width": 1280, "height": 720})
        page = await context.new_page()
        
        # Navigate to the app
        await page.goto("http://localhost:3000/ui/")
        await page.wait_for_load_state("networkidle")
        await asyncio.sleep(2)
        
        # Take initial screenshot
        await page.screenshot(path=str(screenshots_dir / "search_01_initial.png"))
        
        # Find and click on search box
        selector = "input[placeholder*='search' i], input[aria-label*='search' i], .search input, #search"
        search_input = page.locator(selector).first
        if await search_input.count() > 0:
            await search_input.click()
            await asyncio.sleep(0.5)
            await page.screenshot(path=str(screenshots_dir / "search_02_click_search.png"))
            
            # Type search query
            await search_input.type("error", delay=100)
            await asyncio.sleep(1)
            await page.screenshot(path=str(screenshots_dir / "search_03_typing.png"))
            
            # Wait for results
            await asyncio.sleep(2)
            await page.screenshot(path=str(screenshots_dir / "search_04_results.png"))
        else:
            print("Could not find search input")
            # Take a few more screenshots anyway
            await asyncio.sleep(1)
            await page.screenshot(path=str(screenshots_dir / "search_02_nav.png"))
            await asyncio.sleep(1)
            await page.screenshot(path=str(screenshots_dir / "search_03_explore.png"))
        
        await browser.close()
        print(f"Screenshots saved to {screenshots_dir}")

asyncio.run(capture_screenshots())

File: docs/assets/gifs/screenshots/capture_search.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const screenshotsDir = '/home/nistrator/Documents/github/amplifier/ai_working/agent_debugger/docs/assets/gifs/screenshots';
  
  const browser = await chromium.launch({
    headless: false,
    args: ['--disable-dev-shm-usage']
  });
  
  const context = await browser.newContext({
    viewport: { width: 1280, height: 720 }
  });
  
  const page = await context.newPage();
  
  // Navigate to the app
  await page.goto('http://localhost:3000/ui/');
  await page.waitForLoadState('networkidle');
  await page.waitForTimeout(2000);
  
  // Take initial screenshot
  await page.screenshot({ path: path.join(screenshotsDir, 'search_01_initial.png') });
  console.log('Captured: search_01_initial.png');
  
  // Try to find and interact with search
  try {
    // Look for search input
    const searchSelectors = [
      'input[placeholder*="search" i]',
      'input[aria-label*="search" i]',
      '.search input',
      '#search',
      '[data-testid="search"]'
    ];
    
    for (const selector of searchSelectors) {
      const count = await page.locator(selector).count();
      if (count > 0) {
        console.log(`Found search input with selector: ${selector}`);
        await page.locator(selector).first().click();
        await page.waitForTimeout(500);
        await page.screenshot({ path: path.join(screenshotsDir, 'search_02_click_search.png') });
        console.log('Captured: search_02_click_search.png');
        
        // Type search query
        await page.locator(selector).first().type('error', { delay: 100 });
        await page.waitForTimeout(1000);
        await page.screenshot({ path: path.join(screenshotsDir, 'search_03_typing.png') });
        console.log('Captured: search_03_typing.png');
        
        // Wait for results
        await page.waitForTimeout(2000);
        await page.screenshot({ path: path.join(screenshotsDir, 'search_04_results.png') });
        console.log('Captured: search_04_results.png');
        break;
      }
    }
  } catch (e) {
    console.log('Search interaction failed:', e.message);
  }
  
  await browser.close();
  console.log('Screenshots captured successfully');
})();
